src/lib.rs

```diff
@@ -118,40 +128,55 @@ mod test {
         // create mock server
         let server = MockServer::start();
         let ra_mock = server.mock(|when, then| {
-            when.method(GET).path("/");
+            when.method(GET).path("/").header("RA", "true");
             then.status(200)
                 .header("Retry-After", ra_test_duration.to_string())
                 .body("");
         });
         let no_ra_mock = server.mock(|when, then| {
+            when.method(GET).path("/");
+            then.status(200).body("");
+        });
+        let normal_mock = server.mock(|when, then| {
             when.method(GET).path("/normal");
             then.status(200).body("");
         });
 
-        // create request
+        let url = Url::from_str(&server.url("/")).unwrap();
+
+        // hit URL; get RA value and store it
         let now = SystemTime::now();
-        client.get(server.url("/")).send().await.unwrap();
+        client
+            .get(url.clone())
+            .header("RA", "true")
+            .send()
+            .await
+            .unwrap();
         ra_mock.assert_async().await;
+        test_some_retry_after(&middleware, &url).await;
+        test_valid_retry_after(&middleware, &url, now, ra_test_duration).await;
 
-        test_some_retry_after(&middleware).await;
-        test_valid_retry_after(&middleware, now, ra_test_duration).await;
-
-        // verify that we actually sleep for the duration of the retry-after header
+        // hit same URL with stored RA; this should (1) sleep and (2) clear the stored RA afterward
+        // meanwhile, hit other URL, which should return instantly
         let now = SystemTime::now();
         client.get(server.url("/normal")).send().await.unwrap();
+        client.get(url.clone()).send().await.unwrap();
+        normal_mock.assert_async().await;
```

Comment:
Hmm, I could do with another duration check here, just to ensure `GET /normal` returned without delay.